test/test_framers.py

```diff
@@ -126,12 +157,26 @@ def test_populate_result(rtu_framer):
     assert result.unit_id == 255
 
 
-@pytest.mark.parametrize('framer', [ascii_framer, rtu_framer, binary_framer])
-def test_process_incoming_packet(framer):
-    def cb(res):
-        return res
-    # data = b''
-    # framer.processIncomingPacket(data, cb, unit=1, single=False)
+@pytest.mark.parametrize("data", [
+    (b'\x11', 17, False, False),  # not complete frame
+    (b'\x11\x03', 17, False, False),  # not complete frame
+    (b'\x11\x03\x06', 17, False, False),  # not complete frame
+    (b'\x11\x03\x06\xAE\x41\x56\x52\x43', 17, False, False),  # not complete frame
+    (b'\x11\x03\x06\xAE\x41\x56\x52\x43\x40', 17, False, False),  # not complete frame
+    (b'\x11\x03\x06\xAE\x41\x56\x52\x43\x40\x49', 17, False, False),  # not complete frame
+    (b'\x11\x03\x06\xAE\x41\x56\x52\x43\x40\x49\xAC', 17, True, False),  # bad crc
+    (b'\x11\x03\x06\xAE\x41\x56\x52\x43\x40\x49\xAD', 17, False, True),  # good frame
+    (b'\x11\x03\x06\xAE\x41\x56\x52\x43\x40\x49\xAD', 16, True, False),  # incorrect unit id
+    (b'\x11\x03\x06\xAE\x41\x56\x52\x43\x40\x49\xAD\x11\x03', 17, False, True),  # good frame + part of next frame
+])
+def test_rtu_process_incoming_packet(rtu_framer, data):
```

Comment:
Improve coverage and ensure that RTU framer properly deals with incomplete frames